connectors/apache-connector/src/main/java/org/glassfish/jersey/apache/connector/ApacheConnector.java

```diff
@@ -648,7 +636,6 @@ private static InputStream getInputStream(final CloseableHttpResponse response)
             @Override
             public void close() throws IOException {
                 response.close();
-                super.close();
             }
```

Comment:
@agherardi I agree with you. However `response.close()` will always cause the connection to be closed regardless of whether or not the entity was read in its entirety. Follow the chain of calls that response.close() makes and you will see it eventually calls `ConnectionHolder.releaseConnection(false)` - the `false` bit means the connection should not be reused (returned to the pool).

```
HttpClientResponseInputStream.close() -> CloseableHttpResponse.close() -> HttpResponseProxy.close() -> ConnectionHolder.close() -> ConnectionHolder.releaseConnection(false)
```

There needs to be some conditional check to see if the entity has not been consumed yet and then call `response.close()` otherwise call `super.close()`. 

Another alternative could be to have an option to consume/read any remaining bytes of the entity and discard them, I'm not sure this should be the default behavior but it seems very useful.